src/renderer/ducks/validator/sagas.ts

```diff
@@ -445,17 +446,19 @@ export function* getAttestationEffectiveness({
                 skippedQue++;
             } else {
                 const sanitizedAttestations = result.filter(
-                    ({data}) => payload.block === toHex(data.beaconBlockRoot) && data.index === payload.index,
+                    ({data, aggregationBits}) =>
+                        payload.block === toHex(data.beaconBlockRoot) &&
+                        data.index === payload.index &&
+                        data.slot === payload.slot &&
+                        payload.bits === JSON.stringify(aggregationBits),
```

Comment:
You need to check if bit on validator position in committee is set to true. I think you should also check source and target root.

Reply:
`source` and `target` contains `epoch` and `root` of `slot` as i check `source` `slot` in parent object of `source` and think checking for `source` and `target` can be bit a overkill and add unnecessary complexity

Reply:
it contains epoch and epoch boundary root (it's basically validator vote) that are unreleated to slot. That said since, you are checking committee index, it's highly unlikely that someone created attestation with different source and target outside of chainguardian (in that case more pressing issue is slashing anyways).